az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -352,24 +364,50 @@ public static void proxyUserKillAllSpawnedHadoopJobs(
     final Properties properties = new Properties();
     properties.putAll(jobProps.getFlattened());
 
-    try {
-      if (HadoopSecureWrapperUtils.shouldProxy(properties)) {
-        final UserGroupInformation proxyUser =
-            HadoopSecureWrapperUtils.setupProxyUserWithHSM(hadoopSecurityManager, properties,
-                tokenFile.getAbsolutePath(), log);
-        proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
-          @Override
-          public Void run() throws Exception {
-            findAndKillYarnApps(jobProps, log);
-            return null;
-          }
-        });
-      } else {
-        findAndKillYarnApps(jobProps, log);
+    // todo: use feature flag, default to use legacy mode
+    String yarnKillVersion = jobProps.getString(YARN_KILL_VERSION, YARN_KILL_LEGACY);
+    if (YARN_KILL_USE_API_WITH_TOKEN.equals(yarnKillVersion)){
+      try {
+        if (HadoopSecureWrapperUtils.shouldProxy(properties)) {
+          final UserGroupInformation proxyUser =
+              HadoopSecureWrapperUtils.setupProxyUserWithHSM(hadoopSecurityManager, properties,
+                  tokenFile.getAbsolutePath(), log);
+          proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
+            @Override
+            public Void run() throws Exception {
+              findAndKillYarnApps(jobProps, log);
+              return null;
+            }
+          });
+        } else {
+          findAndKillYarnApps(jobProps, log);
+        }
+      } catch (final Throwable t) {
+        log.warn("something happened while trying to kill all spawned jobs", t);
+      }
+    } else {
+      final String logFilePath = jobProps.getString(CommonJobProperties.JOB_LOG_FILE);
```

Comment:
Is the entire `else {}` block legacy code from before the current yarn-kill feature was added ?

Reply:
Affirmative, legacy code before my change.